x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/SqlVersion.java

```diff
@@ -159,9 +159,12 @@ public static boolean hasVersionCompatibility(SqlVersion version) {
         return version.compareTo(V_7_7_0) >= 0;
     }
 
-    public static boolean isClientCompatible(SqlVersion version) {
-        /* only client's of version 7.7.0 and later are supported as backwards compatible */
-        return V_7_7_0.compareTo(version) <= 0;
+    // A client is version-compatible with the server if:
+    // - it supports version compatibility (past or on 7.7.0); and
+    // - it's not on a version newer than server's; and
+    // - it's major version is at most one unit behind server's.
+    public static boolean isClientCompatible(SqlVersion server, SqlVersion client) {
```

Comment:
Please add a comment that we need the 2 args because the server version is not available here (not a dependency).

Reply:
Sure, added.